src/redux.js

```diff
@@ -109,6 +111,35 @@ export function getFeatures(state) {
   return state.data.length > 0 ? Object.keys(state.data[0]) : [];
 }
 
+function filterColumnsByType(state, columnType) {
+  let columnsOfType = [];
+  if (!_.isEmpty(state.metaDataByColumn)) {
+    var filteredObject = _.pickBy(state.metaDataByColumn, function(
+      columnMetaData,
+      column
+    ) {
+      return columnMetaData["dataType"] === columnType;
+    });
+    columnsOfType = Object.keys(filteredObject);
+  }
+  return columnsOfType;
+}
+
+export function getCategoricalColumns(state) {
+  return filterColumnsByType(state, ColumnTypes.CATEGORICAL);
+}
+
+export function getContinuousColumns(state) {
+  return filterColumnsByType(state, ColumnTypes.CONTINUOUS);
+}
+
+export function getSelectableFeatures(state) {
+  return _.pull(
```

Comment:
nit: this mutates the array in place, and while possibly more efficient, it might be good to use immutable alternatives whenever possible?

Reply:
It sounds like you're already doing this for most of the data flow, but just to capture the idea: given all the different ways we manipulate the original CSV-based data, it would be good to treat that data as immutable and to always have a chain of transformations that generate new views of that original data, rather than ever modifying the original data.